smack-core/src/main/java/org/jivesoftware/smack/proxy/ProxyInfo.java

```diff
@@ -24,17 +28,18 @@
  */
 
 public class ProxyInfo {
+
     public enum ProxyType {
         HTTP,
         SOCKS4,
         SOCKS5
     }
 
-    private String proxyAddress;
-    private int proxyPort;
-    private String proxyUsername;
-    private String proxyPassword;
-    private ProxyType proxyType;
+    private final String proxyAddress;
```

Comment:
This and the following changes adding `final` seem very sensible, but should be an extra commit and pull request.

Reply:
Alrighty, will do